src/aiida/common/pydantic.py

```diff
@@ -41,7 +41,7 @@ class Model(BaseModel):
     field_info = Field(default, **kwargs)
 
     for key, value in (('priority', priority), ('short_name', short_name), ('option_cls', option_cls)):
-        if value is not None:
+        if value is not None and field_info is not None:
             field_info.metadata.append({key: value})
```

Comment:
I see the error but does this mean the `Field` constructor can return None?

Reply:
just saw https://github.com/aiidateam/aiida-core/pull/6630#discussion_r1854317395 interesting